src/librustc/session/config.rs

```diff
@@ -2422,6 +2427,11 @@ mod dep_tracking {
     }
 }
 
+pub fn edition_name_list() -> String {
```

Comment:
Would really love some guidance here. This function really needs to return a `&'static str` but I'm not sure how to do that. Any thoughts?

Reply:
Must it return a `'static str`?  There's no way to do this without some unsafe code or hardcoding it.

Reply:
So the reason I think it needs to be `'static str` is that on line 1646, `opt::opt_s` requires that the "valid options string" be of type `&'static str`. Is there some way to generate this String while:
  1. Not introducing something that needs to change whenever we add a new edition in the `libsyntax/edition.rs` file.
  2. And generating a string that is of type `&'static str`